tests/jballerina-unit-test/src/test/resources/test-src/types/uniontypes/union_types_basic.bal

```diff
@@ -166,3 +166,70 @@ function testUnionLhsWithDiscriminatedFloatDecimalLiterals() returns [(float|dec
     float|decimal c = 1.0d;
     return [a, b, c];
 }
+
+type Employee abstract object {
+    public int age;
+    public string firstName;
+    public string lastName;
+
+    function getFullName() returns string;
+};
+
+type Engineer object {
+    public int age;
+    public string firstName;
+    public string lastName;
+
+    function init(int age, string firstName, string lastName) {
+        self.age = age;
+        self.firstName = firstName;
+        self.lastName = lastName;
+    }
+
+    function getFullName() returns string {
+        return self.firstName + self.lastName;
+    }
+
+};
+
+type Manager object {
+    int age;
+    string firstName;
+    string lastName;
+
+    function init(int age, string firstName, string lastName) {
+        self.age = age;
+        self.firstName = firstName;
+        self.lastName = lastName;
+    }
+};
+
+function testUnionTypeWithFunctionPointerAccess() {
+    Engineer engineer = new Engineer(20, "John", "Doe");
+    Employee employee = new Engineer(20, "Jane", "Doe");
+
+    Engineer|Manager person1 = engineer;
+    Employee|Manager person2 = employee;
```

Comment:
Are we not testing Manager type here?

Reply:
It works same as the engineer type. We can remove it and use only the 'employee' and 'engineer' types. Is it the concern?

Reply:
Removed the unnecessary object type in the test case